src/corva/models/stream/raw.py

```diff
@@ -165,33 +168,48 @@ def filter_records(
     def set_asset_id(cls, values: dict) -> dict:
         """Calculates asset_id field."""
 
-        records: List[RawBaseRecord] = values['records']
+        records: List[RawBaseRecord] = values["records"]
 
-        values["asset_id"] = int(records[0].asset_id)
+        if records:
+            values["asset_id"] = int(records[0].asset_id)
 
         return values
 
     @pydantic.root_validator(pre=False, skip_on_failure=True)
     def set_company_id(cls, values: dict) -> dict:
         """Calculates company_id field."""
 
-        records: List[RawBaseRecord] = values['records']
+        records: List[RawBaseRecord] = values["records"]
 
-        values["company_id"] = int(records[0].company_id)
+        if records:
+            values["company_id"] = int(records[0].company_id)
 
         return values
 
+    @pydantic.validator("records", pre=True)
+    def validate_records(cls, v):
+        if isinstance(v, List):
```

Comment:
Is it a good idea long-term to use `typing.List` in `isinstance` checks? Simple `list` might be more robust, especially, if we migrate to 3.9+

Reply:
But this change will break SDK for old Dev Center apps which are still on Python3.8 or am I missing anything?

Reply:
It won't, e.g. below we use simple `dict` in `isinstance` check - same thing can be used here